idea/idea-frontend-api/src/org/jetbrains/kotlin/idea/frontend/api/components/KtOverrideInfoProvider.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.frontend.api.components
+
+import org.jetbrains.kotlin.descriptors.ImplementationStatus
+import org.jetbrains.kotlin.idea.frontend.api.symbols.KtCallableSymbol
+import org.jetbrains.kotlin.idea.frontend.api.symbols.KtClassOrObjectSymbol
+
+abstract class KtOverrideInfoProvider : KtAnalysisSessionComponent() {
+    abstract fun isMemberVisible(memberSymbol: KtCallableSymbol, parentClassSymbol: KtClassOrObjectSymbol): Boolean
+    abstract fun getImplementationStatus(memberSymbol: KtCallableSymbol, parentClassSymbol: KtClassOrObjectSymbol): ImplementationStatus?
+    abstract fun getIntersectionSymbols(symbol: KtCallableSymbol): List<KtCallableSymbol>?
```

Comment:
there is already KtSymbolDeclarationOverridesProviderMixIn.getIntersectionOverriddenSymbols

Reply:
Ah! Let me us that!